src/app.js

```diff
@@ -0,0 +1,194 @@
+import axios from 'axios';
+
+
+// Makes an identifier variable for the countryInformation div.
+const countryInformation = document.getElementById('countryInformation');
+
+
+//Makes the searchbar and button in a div together
+const searchBar = document.createElement("div");
+searchBar.setAttribute('style', 'border: 1px solid black;')
+searchBar.innerHTML = `
+                <input type="text" id="searchText"/>
+                <button id='submitSearch' style="background-color: green">Submit</button>
+             `;
+
+
+// Makes sure the form doesn't change back to default after submitting.
+const preventDefault = (e) => {
+    e.preventDefault();
+}
+countryInformation.addEventListener('click', preventDefault)
+
+//Puts the searchbar in the countryInformation form.
+countryInformation.appendChild(searchBar);
+
+// Makes identifier variables of the above created submit and searchbar.
+const submitSearch = document.getElementById('submitSearch');
+const searchText = document.getElementById('searchText');
+
+
+// The async function that fetches data through an api.
+async function getCountries() {
+    try {
+
+        //Gets data
+        const result = await axios.get('https://restcountries.com/v2/all');
+
+        // Puts the array with objects we got in a variable
+        const allData = result.data;
+
+        //An indexNumber to identify a specific object in this case a country!
+        let indexNumber;
+
+
+        // Function that is being called when user puts something in the search field!
+        // If the user input exists it will connect the input to the index number, identifying the object array[indexNumber].
+        const search = (e) => {
+           // returns the names of the objects
+            const names = allData.map((item) => {
+              return item.name
+            })
+            //Checks if the input name exists in the array object, if it does then it returns the indexNumber of the array where the object is located in.
+            if (names.includes(e.target.value)) {
+                   for (let i = 0; i < names.length; i++) {
+                       if (names[i] === e.target.value) {
+                           indexNumber = i;
+                       }
+                   }
+               }
+
+               //If the name does not exist it will return false!
+               else {
+                   indexNumber = false;
+               }

```

Comment:
Volgens mij kun je ook `indexNumber = allData.findIndex(({ name }) => name.toLowerCase() === e.target.value.toLowerCase());` doen. Dan komt er vanzelf -1 uit als de waarde niet gevonden kan worden. Let ook op dat ik `.toLowerCase()` heb toegevoegd zodat het niet uitmaakt of je hoofdletters gebruikt of niet.

Reply:
Ik zie dat je met deze index altijd opnieuw het land ophaalt, kun je dan niet beter `const foundCountry = allData.find(({ name }) => name.toLowerCase() === e.target.value.toLowerCase());` doen?